sdk/helper/storagepacker/ids2keys.go

```diff
@@ -0,0 +1,106 @@
+package storagepacker
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/vault/sdk/helper/cryptoutil"
+	"math"
+	"sort"
+	"strings"
+)
+
+type itemRequest struct {
+	// Item ID, provided by client
+	ID string
+
+	// Storage key == hash of ID
+	Key string
+
+	// Stored object, nil if not found
+	Value *Item
+}
+
+func GetItemIDHash(itemID string) string {
+	return hex.EncodeToString(cryptoutil.Blake2b256Hash(itemID))
+}
+
+// Given a list of IDs, calculate their keys generate itemRequests for each.
+func (s *StoragePackerV2) keysForIDs(ids []string) []*itemRequest {
+	requests := make([]*itemRequest, 0, len(ids))
+	for _, id := range ids {
+		requests = append(requests, &itemRequest{
```

Comment:
I might be wrong, but I think here it actually needs to be:
```
	requests := make([]*itemRequest, 0, len(ids))
	for i, id := range ids {
		requests[i] = &itemRequest{
```
At least in previous versions of Go, it ended up appending to the _end_ of the loop above and extending its size, rather than inserting an element at the intended index.

Reply:
Oh, wait, no, ignore me - that's with the other method of making a slice.